components/brave_ads/core/internal/user/user_interaction/ad_events/ad_events_database_table.cc

```diff
@@ -309,6 +309,20 @@ void AdEvents::PurgeOrphaned(const mojom::AdType ad_type,
   RunTransaction(std::move(transaction), std::move(callback));
 }
 
+void AdEvents::PurgeAllOrphaned(ResultCallback callback) const {
+  mojom::DBTransactionInfoPtr transaction = mojom::DBTransactionInfo::New();
+  mojom::DBCommandInfoPtr command = mojom::DBCommandInfo::New();
+  command->type = mojom::DBCommandInfo::Type::EXECUTE;
+  command->sql = base::ReplaceStringPlaceholders(
+      "DELETE FROM $1 WHERE placement_id IN (SELECT placement_id from $2 "
+      "GROUP BY placement_id having count(*) = 1) AND confirmation_type IN "
+      "(SELECT confirmation_type from $3 WHERE confirmation_type = 'served');",
```

Comment:
I think this cab be simplified to `confirmation_type = 'served'`

Reply:
Discussed and will be looked at when we optimize queries in another open issue

Reply:
Discussed and database queries will be optimized in follow up issues